Board.java

```diff
@@ -1,97 +1,46 @@
-      }
-      if(location == 51){
-         System.out.println("Congratulations, you landed on a ladder taking you from " + location + " to " + 67);
-         return 67;
-      }
-      if(location == 56){
-         System.out.println("Uh oh! You landed on a chute taking you from " + location + " to " + 53);
-         return 53;
-      }
-      if(location == 62){
-         System.out.println("Uh oh! You landed on a chute taking you from " + location + " to " + 19);
-         return 19;
-      }
-      if(location == 64){
-         System.out.println("Uh oh! You landed on a chute taking you from " + location + " to " + 60);
-         return 60;
-      }
-      if(location == 71){
-         System.out.println("Congratulations, you landed on a ladder taking you from " + location + " to " + 91);
-         return 91;
-      }
-      if(location == 80){
-         System.out.println("Congratulations, you landed on a ladder taking you from " + location + " to " + 100);
-         return 100;
-      }
-      if(location == 87){
-         System.out.println("Uh oh! You landed on a chute taking you from " + location + " to " + 24);
-         return 24;
-      }
-      if(location == 93){
-         System.out.println("Uh oh! You landed on a chute taking you from " + location + " to " + 73);
-         return 73;
-      }
-      if(location == 95){
-         System.out.println("Uh oh! You landed on a chute taking you from " + location + " to " + 75);
-         return 73;
-      }
-      if(location == 98){
-         System.out.println("Uh oh! You landed on a chute taking you from " + location + " to " + 78);
-         return 73;
-      }
-      else{
-         int newLocation = location + spin;
-         while(newLocation > 100){
-            newLocation --;
-         }
-         System.out.println("You advance from " + location + " to " + newLocation);
-         return newLocation;
-      }
+   private static final int[][] LADDERS = {
+     {1, 38}, {4, 14}, {9, 31}, {21, 42}, {28, 84}, {36, 44}, {51, 67}, {71, 91}, {80, 100}};
+   private static final int[][] CHUTES = {
+     {16, 6}, {47, 26}, {49, 11}, {56, 53}, {62, 19}, {64, 60}, {87, 24}, {93, 73}, {95, 73}, {98, 73}};
+   private static String CONGRATS = "Congratulations, you landed on a ladder taking you from ";
+   private static String UH_OH = "Uh oh! You landed on a chute taking you from ";
+   private static String TO = " to ";
+
+   private static int adjustedNewPositionWith(
+      int[][] shortcutsData, int position)
+   {
+      int newPosition = 0;
+      for (int i=0; i<shortcutsData.length; i++)
+      {
+         if (position == shortcutsData[i][0])
+         { newPosition = shortcutsData[i][1]; break; }
+      }
+      return newPosition;
+   }
+
+   public static int getNewLocation(int currentPosition, int spinnerValue)
+   {
+      int newPositionAfterSpin = currentPosition + spinnerValue;
+      int effectiveNewPosition = adjustedNewPositionWith(LADDERS, newPositionAfterSpin);

```

Comment:
since i had 2 substitute `null` here the logic needs some explanation:
- new 'effective' position is calculated based on the 'proposed' new position (current + number on the spinner)
- if there's no chute nor ladder @ the 'proposed' new position then the result of the 'adjusted' position method is `0`. 
- if that 'adjusted' position is `0` it means the player simply goes to the 'proposed' new position. but we need 2 checks (since chutes and ladders r separated).

technically, the logic might b different. we can combine both 'ladders' and 'chutes' in a single data structure but instead of 'target' positions we will have positive/negative 'adjustment' values.
then the logic here will be a bit more complicated but potentially more straightforward:

- calculate 'position adjustment' (which will b either positive or negative number or 0 (if there's no chute nor ladder @ the 'proposed' position after spin)
- calculate `effective new position` by simply adding the 'position adjustment'
- then print the message based on adjustment:
- - if > 0: print 'congrats'
- - if < 0: print 'uh-oh'
- - otherwise print 'regular advancement'.

Reply:
```suggestion
      int positionAdjustment = positionAdjustment(newPositionAfterSpin); // use single data model, no other parameters needed
      if (positionAdjustment > 0)
      { /* print congrats */ }
      else if (positionAdjustemtn < 0)
      { /* print uh-oh*/ }
      else
      { /* print advancement */ }
      return newPositionAfterSpin + positionAdjustment;
```
actually.  this means the code will b even shorter.